R/makeTreeSummarizedExperimentFromqiime2.R

```diff
@@ -0,0 +1,245 @@
+    # if is DNA, row.names(feature_tab) is set as refseq
+    if  (is_dna_seq(row.names(feature_tab))) {
+        refseq <- row.names(feature_tab)
+        refseq_nm <- paste0("OTU", seq_along(refseq))
+        names(refseq) <- refseq_nm
+
+        # set the rownames of feature and taxa tab as OTU1, OTU2,...
+        if (!is.null(taxa_tab)) {
+            rownames(taxa_tab) <- refseq_nm
+        }
+        rownames(feature_tab) <- refseq_nm
+    } else {
+        refseq <- NULL
+    }
+
+    if (!is.null(refseq_qza)){
+        refseq <- read_qza(refseq_qza)
+    }
+
+
+    if(!is.null(refseq)) {
+        refseq <- Biostrings::DNAStringSet(refseq)
+    } else {
+        refseq <- NULL
+    }
+
+    TreeSummarizedExperiment(
+        assays = S4Vectors::SimpleList(counts = feature_tab),
+        rowData = taxa_tab,
+        colData = sam_tab,
+        rowTree = tree,
+        referenceSeq = refseq
+    )
+}
+
+
+#' Read the qza file output from qiime2
+#'
+#' Import the qiime2 artifacts to R.
+#'
+#' @param file character, path of the input qza file. Only files in format of
+#'   `BIOMV210DirFmt` (feature table), `TSVTaxonomyDirectoryFormat` (taxonomic
+#'   table), `NewickDirectoryFormat` (phylogenetic tree ) and
+#'   `DNASequencesDirectoryFormat` (representative sequences) are supported
+#'    right now.
+#' @param temp character, a temporary directory in which the qza file will be
+#'   decompressed to, default `tempdir()`.
+#' @return [`phyloseq::otu_table-class`] object for feature table,
+#'   [`phyloseq::taxonomyTable-class`] object for taxonomic table,
+#'   [`ape::phylo`] object for phylogenetic tree,
+#'   [`Biostrings::DNAStringSet-class`] for representative sequences of taxa.
+#' @noRd
+#' @keywords internal
+read_qza <- function(file, temp = tempdir()) {
+    unzipped_file <- utils::unzip(file, exdir = temp)
+    meta_file <- grep("metadata.yaml", unzipped_file, value = TRUE)
+    metadata <- yaml::read_yaml(meta_file[1])
+    format <- metadata$format
+    uuid <- metadata$uuid
+
+    if (grepl("BIOMV", metadata$format)) {
+        biom_file <- file.path(temp, uuid,"data/feature-table.biom")
+        res <- read_q2biom(biom_file)
+    } else if (format == "TSVTaxonomyDirectoryFormat") {
+        taxa_file <- file.path(temp, uuid, "data/taxonomy.tsv")
+        res <- read_q2taxa(taxa_file)
+    } else if (format == "NewickDirectoryFormat") {
+        tree_file <- file.path(temp, uuid, "data/tree.nwk")
+        res <- ape::read.tree(tree_file)
+    } else if (format == "DNASequencesDirectoryFormat") {
+        refseq_file <- file.path(temp, uuid, "data/dna-sequences.fasta")
+        res <- Biostrings::readDNAStringSet(refseq_file)
+    } else {
+        stop(
+            "Only files in format of 'BIOMV210DirFmt', ",
+            "'TSVTaxonomyDirectoryFormat', NewickDirectoryFormat' and ",
+            "'DNASequencesDirectoryFormat' are supported."
+        )
+    }
+
+    res
```

Comment:
Use a switch statement (and maybe direct to individual wrapper functions and exectue with `do.call`)